lib/elements/clone.py

```diff
@@ -86,7 +86,7 @@ def to_stitch_groups(self, last_patch=None):
         patches = []
 
         source_node = get_clone_source(self.node)
-        if source_node.tag not in EMBROIDERABLE_TAGS:
+        if source_node.tag not in EMBROIDERABLE_TAGS and source_node.tag != SVG_GROUP_TAG:
```

Comment:
Since we are now using groups now, we need to do the angle (not the transform) modifications inside the elements loop.

Reply:
Ah, true. I'll fix that.

Reply:
This is actually a bit harder to deal with than I thought. It's possible to apply the angle like so:

```
        for element in elements:
            old_angle = element.get_float_param('angle', 0)
            element.set_param('angle', old_angle + angle)
            element.get_param.cache_clear() # Hack: Invalidate the cache so that the updated angle will be returned.
            element.get_float_param.cache_clear() # Hack: Invalidate the cache so that the updated angle will be returned.
            stitch_groups = element.to_stitch_groups(last_patch)
            patches.extend(stitch_groups)
            element.set_param('angle', old_angle)
```

but then you have the problem that you get old, cached versions of the `angle` attribute from the element because fetching it the first time caches it. And we have to reset the angle value on the element because we operate on it in-place.

![image](https://github.com/inkstitch/inkstitch/assets/5505503/40d8972a-9ee8-4217-9221-67c7129e528b)

Might be easier to duplicate the href'd elements of a clone, though naively `copy.deepcopy`ing `source_node` leads to the following:

![image](https://github.com/inkstitch/inkstitch/assets/5505503/10b7c1e4-1c9c-493c-aaca-8f4a1d0ef174)

I suppose I'll take another look at it tomorrow.

Reply:
If you find a better way, you are most welcome to improve the angle rotation code as a whole. The whole transform part is a bit odd codewise, but such a very useful feature to have for clones. Sometimes you even need to active the flip option to get it right. It'd be great if this wouldn't be necessary.

I may be wrong, but if you feed the _get_rotation method with a group it may fail in case it can't read the rotation directly from the transform? This means you either have to recalculate for every element within the group or make sure to feed it with path elements to compare beforehand. But maybe you did that already. ... Or rewrite it completely :D

Reply:
See, I was trying to avoid the classic "I don't immediately understand this code and maybe have one or two Opinions on how it's written, time to completely rewrite it my way" pitfall, but if I have license to do so... 

Definitely a case where it'd be nice to have unit tests implemented. I didn't want to step on the toes of the in-flight PR(s) adding those yet, though.